hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/PlanExecutor.java

```diff
@@ -496,12 +503,59 @@ SqlResult execute(ExplainStatementPlan plan) {
 
     SqlResult execute(SelectPlan plan, QueryId queryId, List<Object> arguments, long timeout) {
         List<Object> args = prepareArguments(plan.getParameterMetadata(), arguments);
+        InternalSerializationService serializationService = Util.getSerializationService(hazelcastInstance);
+        ExpressionEvalContext evalContext = new ExpressionEvalContextImpl(
+                args,
+                serializationService,
+                Util.getNodeEngine(hazelcastInstance));
+
         JobConfig jobConfig = new JobConfig()
                 .setArgument(SQL_ARGUMENTS_KEY_NAME, args)
                 .setArgument(KEY_SQL_QUERY_TEXT, plan.getQuery())
                 .setArgument(KEY_SQL_UNBOUNDED, plan.isStreaming())
                 .setTimeoutMillis(timeout);
 
+        final Set<Integer> partitions = new HashSet<>();
+        for (Entry<String, Map<String, Expression<?>>> candidate : plan.getPartitionStrategyCandidates().entrySet()) {
+            // TODO: null checks
+            final String mapName = candidate.getKey();
+            final Map<String, Expression<?>> keyExprs = candidate.getValue();
+            final MapContainer mapContainer = getMapContainer(hazelcastInstance.getMap(mapName));
```

Comment:
do we really need to go to such low level (MapContainer)?

Reply:
Is there a better way to get the strategy instance, rather than just its config?

Reply:
In WriteMapP we get if from `((MapProxyImpl<K, V>) map).getPartitionStrategy()`